### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by jayres**
Re-organized the Readme and created the local setup doc. 

It looks like there are some conflicts with the readme.md file but I think this is an overall improvement. Do we want to verify that my directions actually work?

### CODE REVIEW

**Comment by jayres** [comment]
Issue #180

**Comment by oshoham** [comment]
@jayres this is a minor nitpick, but can we still with ALL_CAPS titles for our markdown files for consistency?

File: docs/localsetupinstructions.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+## Setting Up Call Party Locally
+
+### 1. Install System Dependencies
+
+#### OS X
+Install MongoDB, Redis, and NVM
+
+```bash
+brew install mongodb redis
```
Comment (oshoham) [inline docs/localsetupinstructions.md]: we can remove `redis` from this line

File: docs/localsetupinstructions.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash # install nvm
+
+export NVM_DIR="$HOME/.nvm"
+
+[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh" # This loads nvm
+
+nvm install 7.4.0
```
Comment (oshoham) [inline docs/localsetupinstructions.md]: we should change this to either `7.6.0` or `stable`

**Comment by oshoham** [comment]
@jayres how would you feel about a table of contents at the top of the README that links to sections and other markdown files, such as `CONTRIBUTING.md`?

**Comment by jayres** [comment]
Cool. Made a bunch of those changes. I think we can add a table of contents when we have more info on diff docs. At this point there is only two items worth noting and the other two docs in the /docs folder are incomplete at this moment.

File: docs/LOCAL_SETUP_INSTRUCTIONS.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+2. Select: View as Page Visitor
+
+3. Send Message
+
+——
+
+## Notes:
+
+If changes are made locally, you will have to restart the server. Return to 2nd backend window (not the localtunnel process):
```
Comment (mhfowler) [inline docs/LOCAL_SETUP_INSTRUCTIONS.md]: if you are using `npm start` in the backend, I think we are  using nodemon now, which will automatically restart the server if any files are change. So I think we can delete this notes section

**Pull request merged**